### ISSUE TITLE
Handling Similar but Not Identical Names for JSON-to-RDF Conversion in LinkML

### ISSUE THREAD

**Issue by danielbarta** [linkml/linkml#1536]
Dear LinkML Development Team,

I am writing to discuss an issue I encountered while attempting to convert JSON data into RDF using LinkML. I have created a LinkML schema from an OWL ontology using Python, and I would like to utilize this schema for the JSON to RDF conversion. However, I have come across a situation where certain arguments in the JSON data do not have an exact match in the schema or ontology, but rather have similar names.

For instance: the JSON file contains an argument named "algorithmId," while the corresponding term in the OWL ontology is defined as "owl:algorithm." Although the names are not identical, they represent the same concept.

My question is whether LinkML provides a method or mechanism to handle such cases, where the names in the JSON data and the schema/ontology are not an exact match, but the underlying concepts align. Is there a recommended approach or feature within LinkML that can assist in resolving this issue?

I would greatly appreciate any guidance or suggestions you can provide to help me address this challenge effectively. Thank you very much for your time and attention to this matter.

Best regards, Daniel

**Comment by turbomam** [comment]
Correct, the names of LinkML schema elements (including the LinkML metamodel itself) are changed to at least be compatible with the various data serializations we support, and preferably to follow the naming best practices of those serializations.

Some of this can be discovered by looking for element URIs. I'll post an example in a minute. But there may be other cases in which understanding the renaming might require looking at the code that generates the serialization. We should have better documentation or reporting for that.

**Comment by turbomam** [comment]
Sorry, maybe I misunderstood your question. I don't see 'algorithm' anywhere in the [LinkML metamodel repository](https://github.com/linkml/linkml-model). Are "algorithmId" and  "owl:algorithm" terms that appear in your input and output from some automated schema building or conversion process? Maybe using the [schema automator](https://github.com/linkml/schema-automator)?

If so, could you paste or link to your input, your building or conversion commands, and your output here in the issue?

**Comment by danielbarta** [comment]
In the process of utilizing the LinkML-convert tool to transform a JSON file to RDF format, I've created a LinkML schema based on a particular ontology. However, I've come across a difficulty wherein the key names in my JSON file do not align with those defined within the ontology and consequently, the created schema(for example:"algorithm" and "algorithmId"). Is there a feature within LinkML that allows for the mapping of these key names in my JSON file to the appropriate classes defined in the ontology and schema?

**Comment by sierra-moxon** [comment]
while still very much in development, you could check out https://github.com/linkml/linkml-transformer

**Comment by sierra-moxon** [comment]
@danielbarta - did we answer your question or would you like to discuss further?  :)

**Comment by cmungall** [comment]
Closing this issue as the question was answered - linkml-map (previously linkml-transformer) is the recommended approach for complex JSON-to-RDF name mappings. No follow-up in over 18 months.

If this is still a problem, please reopen with additional details.

— Claude (reviewed and approved by @cmungall)
